2. Make an API request that returns all the times a query was run more than 100 times in a month. How many times did this occur? 

In [56]:
import requests
import pandas as pd
import matplotlib.pyplot as plt


%matplotlib inline

In [59]:
# endpoint='https://data.nashville.gov/resource/fuaa-r5cm.json?$limit=50000'
# params={
#         "query_text": "codes"
# }
# response = requests.get(endpoint, params = params)

In [72]:
endpoint='https://data.nashville.gov/resource/fuaa-r5cm.json'
params={
        "$where": "query_count>100",
        "$limit":"50000"
}
response = requests.get(endpoint, params = params)

In [73]:
response

<Response [200]>

In [74]:
response.text

'[{"month_name":"March","year":"2014","query_count":"101","query_text":"permits"}\n,{"month_name":"January","year":"2015","query_count":"101","query_text":"criminal court clerk"}\n,{"month_name":"September","year":"2015","query_count":"101","query_text":"codes"}\n,{"month_name":"March","year":"2016","query_count":"101","query_text":"police"}\n,{"month_name":"March","year":"2016","query_count":"101","query_text":"civil service"}\n,{"month_name":"November","year":"2016","query_count":"101","query_text":"jobs"}\n,{"month_name":"November","year":"2017","query_count":"101","query_text":"metro holidays"}\n,{"month_name":"November","year":"2017","query_count":"101","query_text":"longevity pay"}\n,{"month_name":"January","year":"2018","query_count":"101","query_text":"West Nashville Heights Church of Christ"}\n,{"month_name":"January","year":"2018","query_count":"101","query_text":"property maps"}\n,{"month_name":"March","year":"2018","query_count":"101","query_text":"public works"}\n,{"month_

In [75]:
res = response.json()
res

[{'month_name': 'March',
  'year': '2014',
  'query_count': '101',
  'query_text': 'permits'},
 {'month_name': 'January',
  'year': '2015',
  'query_count': '101',
  'query_text': 'criminal court clerk'},
 {'month_name': 'September',
  'year': '2015',
  'query_count': '101',
  'query_text': 'codes'},
 {'month_name': 'March',
  'year': '2016',
  'query_count': '101',
  'query_text': 'police'},
 {'month_name': 'March',
  'year': '2016',
  'query_count': '101',
  'query_text': 'civil service'},
 {'month_name': 'November',
  'year': '2016',
  'query_count': '101',
  'query_text': 'jobs'},
 {'month_name': 'November',
  'year': '2017',
  'query_count': '101',
  'query_text': 'metro holidays'},
 {'month_name': 'November',
  'year': '2017',
  'query_count': '101',
  'query_text': 'longevity pay'},
 {'month_name': 'January',
  'year': '2018',
  'query_count': '101',
  'query_text': 'West Nashville Heights Church of Christ'},
 {'month_name': 'January',
  'year': '2018',
  'query_count': '101',
 

In [76]:
hun_a_month = pd.DataFrame(res)
hun_a_month

month_name  year query_count            query_text month
0         March  2014         101               permits   NaN
1       January  2015         101  criminal court clerk   NaN
2     September  2015         101                 codes   NaN
3         March  2016         101                police   NaN
4         March  2016         101         civil service   NaN
...         ...   ...         ...                   ...   ...
1257  September  2018         750     annual enrollment   NaN
1258    October  2018         816     annual enrollment   NaN
1259    January  2019        2646             Nashville     1
1260  September  2019        5327             directory     9
1261  September  2019        5610                  test     9

[1262 rows x 5 columns]

In [77]:
hun_a_month['query_count']=hun_a_month['query_count'].astype(int)

In [78]:
hun_a_month['query_count'].unique()

array([ 101,  102,  103,  104,  105,  106,  107,  108,  109,  110,  111,
        112,  113,  114,  115,  116,  117,  118,  119,  120,  121,  122,
        123,  124,  125,  126,  127,  128,  129,  130,  131,  132,  133,
        134,  135,  136,  137,  138,  139,  140,  141,  142,  143,  144,
        145,  146,  147,  148,  149,  150,  151,  152,  153,  154,  155,
        156,  157,  158,  159,  160,  161,  162,  163,  164,  165,  166,
        167,  168,  169,  170,  171,  172,  173,  174,  175,  176,  177,
        178,  179,  180,  181,  182,  183,  184,  185,  186,  187,  188,
        189,  190,  191,  192,  193,  194,  195,  196,  197,  198,  199,
        200,  201,  202,  203,  204,  205,  206,  207,  208,  209,  211,
        212,  213,  215,  217,  219,  221,  222,  223,  224,  225,  226,
        227,  229,  230,  231,  234,  235,  236,  237,  238,  239,  240,
        241,  242,  243,  244,  245,  246,  247,  250,  251,  252,  253,
        254,  255,  256,  264,  265,  266,  267,  2

In [79]:
hun_a_month['year'].unique()

array(['2014', '2015', '2016', '2017', '2018', '2019', '2020', '2021'],
      dtype=object)

In [80]:
hun_a_month.shape

(1262, 5)

there are 1262 instinces where a query was run more than 1000 times a month